pkg/aks/components/Import.vue

```diff
@@ -0,0 +1,129 @@
+<script lang="ts">
+import { defineComponent } from 'vue';
+import { _CREATE } from '@shell/config/query-params';
+import { getAKSClusters } from '../util/aks';
+import LabeledSelect from '@shell/components/form/LabeledSelect.vue';
+import Banner from '@components/Banner/Banner.vue';
+import Checkbox from '@components/Form/Checkbox/Checkbox.vue';
+
+export default defineComponent({
+  name: 'AKSImport',
+
+  emits: ['update:clusterName', 'update:resourceGroup', 'update:resourceLocation', 'update:enableNetworkPolicy'],
+
+  components: {
+    LabeledSelect, Banner, Checkbox
+  },
+
+  props: {
+    mode: {
+      type:    String,
+      default: _CREATE
+    },
+
+    clusterName: {
+      type:    String,
+      default: ''
+    },
+
+    resourceLocation: {
+      type:    String,
+      default: ''
+    },
+
+    resourceGroup: {
+      type:    String,
+      default: ''
+    },
+
+    enableNetworkPolicy: {
+      type:    Boolean,
+      default: false
+    },
+
+    azureCredentialSecret: {
+      type:    String,
+      default: ''
+    },
+
+    rules: {
+      type:    Object,
+      default: () => {
+        return {};
+      }
+    },
+  },
+
+  data() {
+    return {
+      allClusters: [], loadingClusters: false, rbacEnabled: true
+    };
+  },
+
+  watch: {
+    azureCredentialSecret: {
+      handler(neu) {
+        if (neu) {
+          this.getClusters();
+        }
+      },
+      immediate: true
+    }
+  },
+
+  methods: {
+    async getClusters() {
+      this.loadingClusters = true;
+      this.allClusters = await getAKSClusters(this.$store, this.azureCredentialSecret);
+      this.loadingClusters = false;
+    },
```

Comment:
If `getAKSClusters` fails, `this.loadingClusters = false` won't be executed. We should probably wrap this in a try/catch block.

Reply:
Good call - I've updated the method to use a try/catch block and emit an error if `getAksClusters` fails